compiler/src/dotty/tools/dotc/semanticdb/ExtractSemanticDB.scala

```diff
@@ -112,6 +116,15 @@ class ExtractSemanticDB extends Phase:
             case tree: Typed => () // hack for inline code
             case tree        => traverse(tree)
 
+
+    // guarded entrypoints
+    private inline def excludeSymbol0(sym: Symbol)(using Context) = !sym.exists || excludeSymbol(sym)
```

Comment:
Checking for symbol existence is pretty fast, so if this is needed in some cases, I would just add the sym.exists check in the original methods.

Reply:
I did it this way because the methods that are being guarded are often chained as fallbacks, this would reduce redundant checks, but agreed it would be simpler to just put the checks into the methods

Reply:
Yeah I wouldn't worry about the performance impact of redundant exists check

Reply:
Thanks, I've inlined those checks